### PR TITLE
availability map updates

### PR DESCRIPTION
**Pull request by crcro**
two new options in global settings:

- show sysname instead of hostname/ip
- sort by hostname or status

as per requested in #4155

### CODE REVIEW

**Comment by laf** [comment]
Sorry needs a rebase :(

As for sysName, we have something already that deals with that.

ip_to_sysname()

What they are saying is to use sysName if the 'hostname' is an IP.

**Comment by crcro** [comment]
ip_to_sysname is linked with config.php variable force_ip_to_sysname, my change is not based on that

**Comment by laf** [comment]
Yeah it's not linked but it should be, that was my point. People who want ip's to be sysName should only have to set this once.

File: html/includes/common/availability-map.inc.php [hunk 1]
```diff
@@ -93,6 +101,18 @@
                 $host_down_count++;
             }
 
+            if ($config['force_ip_to_sysname'] === true) {
```
Comment (laf) [inline html/includes/common/availability-map.inc.php]: I'm not really sure what all of this if/elseif/else statement is doing but surely you can just do:

`$deviceDisplay = ip_to_sysname($device['hostname']);`

File: html/includes/common/availability-map.inc.php [hunk 2]
```diff
@@ -138,6 +158,18 @@
                     $service_down_count++;
                 }
 
+                if ($config['force_ip_to_sysname'] === true) {
```
Comment (laf) [inline html/includes/common/availability-map.inc.php]: Same here.

**Comment by crcro** [comment]
@laf ... you're right, updated

**Comment by f0o** [comment]
This might be out of scope; currently the availability map folds long hostnames - is there any way to prevent that?

**Comment by florianbeer** [comment]
@f0o easy via CSS `white-space: nowrap;`

https://developer.mozilla.org/en-US/docs/Web/CSS/white-space

**Comment by laf** [comment]
The css won't help (and doesn't), it just means it doesn't wrap but still falls outside of the boxes.

You need to use shorthost() to shorten the hostname to the correct amount. You can specify the length but I've found some of mine still go outside the box so you might have to force a max size still.

**Comment by crcro** [comment]
i think that now all the requirements/recommendations are  implemented

File: sql-schema/131.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO `config` (`config_name`, `config_value`, `config_default`, `config_descr`, `config_group`, `config_group_order`, `config_sub_group`, `config_sub_group_order`, `config_hidden`, `config_disabled`) VALUES('webui.old_availability_map', 'false', 'false', 'Availability map old view', 'webui', 0, 'graph', 0, '1', '0');
```
Comment (laf) [inline sql-schema/131.sql]: You've already added this in sql-schema/129.sql

Once that's removed then we're good for a merge.

Comment (crcro) [inline sql-schema/131.sql]: missed that ... fixed with latest push

**Pull request merged**